VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -25,17 +25,900 @@ if ($PSEdition -eq 'Desktop') {
+                } else {
+                    Write-Warning "Lockdown mode for ESXi host $esxiHost is already set to lockdownNormal: SKIPPED"
+                }
+            }
+        } 
+        
+        if ($PSBoundParameters.ContainsKey("disable")) {
+            Write-Host "Disabling lockdown mode for each ESXi host in $cluster cluster."
+            foreach ($esxiHost in $esxiHosts) {
+                $currentLockdownMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                if ($currentLockdownMode -ne "lockdownDisabled") {
+                    ($esxiHost | Get-View).ExitLockdownMode()
+                    Write-Host "Changing lockdown mode for ESXi host $esxiHost from $currentLockdownMode to lockdownDisabled."
+                    $newLockdownMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                    if ($currentLockdownMode -eq $newLockdownMode) {
+                        Write-Error "Unable to change lockdown mode for ESXi host $esxiHost from $currentLockdownMode to lockdownDisabled. Lockdown mode is set to $newLockdownMode." -ErrorAction Stop
+                    }
+                } else {
+                    Write-Warning "Lockdown mode for ESXi host $esxiHost is already set to lockdownDisabled: SKIPPED"
+                }
+            }
+        } 
+    }
+    Catch {
+        Debug-ExceptionWriter -object $_
+    }
+}
+
+Export-ModuleMember -Function Set-ESXiLockdownMode
+
+
+Function Restart-ESXiHost {
+
+    <#
+        .SYNOPSIS
+        Restart the provided ESXi host and poll for it to come back online
+
+        .DESCRIPTION
+        This cmdlet triggers a restart the provided ESXi host and polls for it to come back online. 
+        Timeout value is in seconds.
+
+        .EXAMPLE
+        Restart-EsxiHost -esxiFqdn sfo01-m01-esx03.sfo.rainpole.io -user root -pass VMw@re1! -poll $true -timeout 1800 -pollInterval 30
+        This example restarts the provided esxi hosts and polls every 30 seconds till it comes back online. It will timeout after 1800 seconds. 
+    #>
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $esxiFqdn,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [bool] $poll = $true,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String] $timeout = 1800,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String] $pollInterval = 30
+    )
+
+    # Connect to ESXi host
+    Connect-VIServer $esxiFqdn -User $user -password $pass -Force
+    Write-Host "Restarting $esxiFqdn"
+    $vmHost = Get-VMHost -Server $esxiFqdn
+    if (!$vmHost) {
```

Comment:
I think that there should be an `} else {` here and then move `Write-Host "Restarting $esxiFqdn"` to there so it's only displayed if the host is found.